lib/services/vote_animation_service.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:flutter/material.dart';
+import 'package:pr0gramm/constants/vote_constants.dart';
+import 'package:pr0gramm/entities/enums/vote.dart';
+import 'package:pr0gramm/entities/enums/vote_button_type.dart';
+import 'package:pr0gramm/views/post/widgets/vote_button.dart';
+
+enum VoteAnimation {
+  initial,
+  voted,
+  focused,
+  unfocused,
+  voteFocused,
+  voteUnfocused,
+  clearFocused,
+  clearUnfocused,
+}
+
+typedef VoteItemFn = Future Function(Vote vote);
+typedef VoteStateChangedHandler = void Function(VoteAnimation voteAnimation);
+
+class VoteAnimationService {
+  final VoteItemFn voteItemHandler;
+  final Vote initialVote;
+  Vote lastVote;
+
+  VoteAnimationService({this.voteItemHandler, this.initialVote})
+      : lastVote = initialVote;
+
+  final Map<VoteButtonType, ValueNotifier<VoteAnimation>> buttonStates =
+      new Map();
+
+  Color getInitialColor(VoteButtonType type) {
+    VoteAnimation initialState = _reduceState(
+      vote: initialVote,
+      button: type.toVote(),
+      whenFocused: VoteAnimation.focused,
+      whenVoted: VoteAnimation.voted,
+      whenElse: VoteAnimation.unfocused,
+    );
+    return (initialState == VoteAnimation.focused ||
```

Comment:
Too much nested ternary operators. Use a few ifs to make it more readable.

Reply:
fixed with e624eddc269b50d7147e96ebaa0e07351be57a39